draft-ietf-quic-transport.md

```diff
@@ -322,175 +322,356 @@ all field sizes are in bits.  When discussing individual bits of fields, the
+* Octets 17+: Payload (version list, containing 0 or more acceptable versions)
 
-* Encrypted Payload: The remainder of a Regular packet is both authenticated and
-  encrypted once packet protection keys are available.  {{QUIC-TLS}} describes
-  packet protection in detail.  After decryption, the plaintext consists of a
-  sequence of frames, as shown in {{regular-packet-frames}}.  Frames are
-  described in {{frames}}.
+The payload of the Version Negotiation packet is a list of 32-bit versions which
+the server supports, as shown below.
 
 ~~~
  0                   1                   2                   3
  0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
 +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
-|                          Frame 1 (*)                        ...
+|                    Supported Version 1 (32)                 ...
 +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
-|                          Frame 2 (*)                        ...
+|                    Supported Version 2 (32)                 ...
 +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
                                ...
 +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
-|                          Frame N (*)                        ...
+|                    Supported Version N (32)                 ...
 +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
 ~~~
-{: #regular-packet-frames title="Contents of Encrypted Payload"}
+{: #version-negotiation-format title="Version Negotiation Packet"}
+
+A server that generates a version negotiation packet SHOULD include one of the
+reserved version numbers (those following the pattern 0x?a?a?a?a).  This ensures
+that clients are able to correctly handle an unsupported version.
+
+The design of version negotiation permits a server to avoid maintaining state
+for packets that it rejects in this fashion.  However, when the server generates
+a version negotiation packet, it cannot randomly generate a reserved version
+number.  This is because the server is required to include the same value in its
+transport parameters (see {{version-validation}}).  To avoid the selected
+version number changing during connection establishment, the reserved version
+can be generated as a function of values that will be available to the server
+when later generating its handshake packets.
+
+A pseudorandom function that takes client address information (IP and port) and
+the client selected version as input would ensure that there is sufficient
+variability in the values that a server uses.
+
+A client MAY send a packet using a reserved version number.  This can be used to
+solicit a list of supported versions from a server.
+
+See {{version-negotiation}} for a description of the version negotiation
+process.
+
+## Public Reset Packet {#public-reset-packet}
+
+A Public Reset packet is sent by only a server and is used to abruptly terminate
```

Comment:
That was my initial reaction as well.  A client implementation on top of a UDP stack will never receive a packet on a closed connection, because there's no UDP socket open there.  The UDP stack will drop packets before they reach QUIC.  However, a server will have a listening socket open and receive all traffic that's sent to that port -- new connections, existing connections, and stale connections.